commit 7eea430e0b2b1a8e1c82c9f1184833ef110b5895 (refs/changes/35/3535/2)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-03-05T08:18:14+00:00 (1 year, 7 months ago)
    
    TRITON-206 vmapi isn't checking free capacity when resizing non-HVM VMs upwards
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Approved by: Josh Wilsdon <jwilsdon@joyent.com>

diff --git a/lib/common/validation.js b/lib/common/validation.js
index 0a42d1c..f766a4d 100644
--- a/lib/common/validation.js
+++ b/lib/common/validation.js
@@ -1377,14 +1377,6 @@ exports.validateUpdateVmParams = function (vmapi, vm, obj, callback) {
     // manually done
     params.subtask = getSubtask();
 
-    // Validate resize. Not allowed for bhyve or kvm at the moment
-    if (params.subtask === 'resize' &&
-        ['bhyve', 'kvm'].indexOf(vm.brand) !== -1) {
-
-        errs.push(errors.invalidParamErr('brand', 'Cannot resize a ' +
-            vm.brand + ' VM'));
-    }
-
     // Async validations
     var asyncFns = [];
     if (params.alias) {
@@ -1393,11 +1385,15 @@ exports.validateUpdateVmParams = function (vmapi, vm, obj, callback) {
     if (params.billing_id) {
         asyncFns.push(validatePackage);
     }
-    if (params.subtask === 'resize' &&
-        ['bhyve', 'kvm'].indexOf(vm.brand) !== -1) {
-
-        asyncFns.push(validateResize);
-        asyncFns.push(validateCapacity);
+    if (params.subtask === 'resize') {
+        if (['bhyve', 'kvm'].indexOf(vm.brand) === -1) {
+            asyncFns.push(validateResize);
+            asyncFns.push(validateCapacity);
+        } else {
+            // Not allowed for bhyve or kvm at the moment
+            errs.push(errors.invalidParamErr('brand', 'Cannot resize a ' +
+                vm.brand + ' VM'));
+        }
     }
 
     function validateAlias(next) {
